packages/flutter_arc_text/example/lib/main.dart

```diff
@@ -9,6 +9,25 @@ void main() => runApp(const MyApp());
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
 
+  CustomDecorationPainter _drawArc(bool clockwise) =>
+      (Canvas canvas, Rect rect, double finalAngle) {
+        final length = 2 * pi - finalAngle;
+        if (length < 0) return;
+
+        final startAngle =
+            clockwise ? finalAngle - pi / 2 : finalAngle + length + pi / 2;
```

Comment:
We shouldn't see that anymore as you suggested: no more tap dancing with angles (outside of the library, of course).

Reply:
Well I am not sure, if that is fully achievable 🤷  

We could of course change the `CustomDecorationPainter` to provide all angle information (and not just the final angle), e.g. something like this

```
typedef CustomDecorationPainter = void Function(
  Canvas canvas,
  Rect rect,
  double decorationStartingAngle,
  double decorationEndAngle,
);
```

Reply:
Actually I think that if you use the canvas rotation you don't really care about the starting angle (already managed by the canvas).